api/src/main/java/com/alibaba/nacos/api/AbstractNacosExtension.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.api;
+
+import com.alibaba.nacos.api.config.filter.IConfigFilter;
+
+import java.util.Properties;
+
+/**
+ * Config Filter Interface.
+ *
+ * @author luyanbo(RobberPhex)
+ */
+public abstract class AbstractNacosExtension implements NacosExtension {
```

Comment:
I can't get why add this abstract class. It just an empty implemetation.

Reply:
The thrid-party extension should only extend AbstractNacosExtension, not implements NacosExtension directly.

If Nacos want add a new method(`buildNamingFilter`) to interface `NacosExtension`, AbstractNacosExtension can implement this method with empty body. In this way, the old third-party extension can still work with new version NacosExtension.

Otherwise, third-party extension could not work, because lack of method `buildNamingFilter`.